app/assets/src/components/SampleUpload.jsx

```diff
@@ -274,21 +292,25 @@ class SampleUpload extends React.Component {
           submitting: false,
           successMessage: "Sample created successfully"
         });
-        setTimeout(() => {
-          this.gotoPage(`/samples/${response.data.id}`);
-        }, 2000);
+        this.goToPageWithTimeout(`/samples/${response.data.id}`);
       })
       .catch(error => {
         this.setState({
           invalid: true,
           submitting: false,
-          serverErrors: error.response.data,
-          errorMessage:
-            "Something went wrong. Try checking if sample name already exists in project?"
+          errorMessage: this.joinServerError(error.response.data)
         });
       });
   }
 
+  joinServerError = function(response) {
```

Comment:
Why do you need this different function declaration?

Reply:
no reason, will change